cdc_kafka/kafka.py

```diff
@@ -280,14 +282,18 @@ def register_schemas(self, topic_name: str, key_schema: Dict[str, Any], value_sc
         if (current_key_schema is None or current_key_schema != key_schema) and not self._disable_writing:
             logger.info('Key schema for subject %s does not exist or is outdated; registering now.', key_subject)
             key_schema_id = self._schema_registry.register(key_subject, key_schema)
-            self._schema_registry.update_compatibility(constants.KEY_SCHEMA_COMPATIBILITY_LEVEL, key_subject)
+            if current_key_schema is None:
+                time.sleep(constants.KAFKA_CONFIG_RELOAD_DELAY_SECS)
+                self._schema_registry.update_compatibility(key_schema_compatibility_level, key_subject)
             registered = True
 
         value_schema_id, current_value_schema, _ = self._schema_registry.get_latest_schema(value_subject)
         if (current_value_schema is None or current_value_schema != value_schema) and not self._disable_writing:
             logger.info('Value schema for subject %s does not exist or is outdated; registering now.', value_subject)
             value_schema_id = self._schema_registry.register(value_subject, value_schema)
-            self._schema_registry.update_compatibility(constants.VALUE_SCHEMA_COMPATIBILITY_LEVEL, value_subject)
+            if current_value_schema is None:
+                time.sleep(constants.KAFKA_CONFIG_RELOAD_DELAY_SECS)
+                self._schema_registry.update_compatibility(value_schema_compatibility_level, value_subject)
```

Comment:
Is there any chance this hasn't finished registering the schema at this point? Can you poll to see if it's ready?

Reply:
We have seen evidence that this can fail sometimes, yes. My speculation is that the request to update compatibility hits a different schema registry process (we run a three-pod deployment in k8s) than the one where the registration was done, and it hasn't yet caught up consuming the topic that the schemas are stored in.

Polling--or at least retrying on failure--is a good idea. Will add to "the list". Thanks!